src/plugin.py

```diff
@@ -1,113 +1,269 @@
-import sys
+import sys, asyncio, logging, urllib, os, json, pickle, webbrowser
+from galaxy.api.plugin import (
+    Plugin,
+    LocalGame,
+    Authentication,
+    Game,
+    create_and_run_plugin,
+    GameTime,
+)
+from galaxy.api.consts import LocalGameState, LicenseType, OSCompatibility, Platform
+from galaxy.api.types import LicenseInfo
 
-from galaxy.api.plugin import Plugin, create_and_run_plugin
-from galaxy.api.consts import Platform
-from galaxy.api.types import Authentication, Game, LicenseInfo, LicenseType, LocalGame, LocalGameState
+from local import WindowsLocalClient, MacLocalClient
+from consts import (
+    GameID,
+    MINECRAFT_DUNGEONS_INSTALL_URL,
+    MINECRAFT_MAC_INSTALL_URL,
+    MINECRAFT_WIN_INSTALL_URL,
+    GAME_NAMES,
+    IS_WINDOWS,
+    INSTALLED_FOLDER_PATH,
+    GAMES,
+)
+from utils import misc, time_tracker
+from utils.decorators import double_click_effect
+import multimc
 from version import __version__
 
-import logging as log
-import os
-import subprocess
-from local import LocalClient
-import asyncio
-import requests
-import tempfile
-import shutil
-from typing import Optional, Any, List
+
+log = logging.getLogger(__name__)
 
 
 class MinecraftPlugin(Plugin):
     def __init__(self, reader, writer, token):
         super().__init__(Platform.Minecraft, __version__, reader, writer, token)
-        self.local_client = LocalClient()
-        self.minecraft_launcher = None
-        self.minecraft_uninstall_command = None
-        self.minecraft_installation_status = LocalGameState.None_
-        self.minecraft_running_check = None
-        self.tick_count = 0
+        self.local_client = WindowsLocalClient() if IS_WINDOWS else MacLocalClient()
+        self.status = {
+            GameID.Minecraft: LocalGameState.None_,
+            GameID.MinecraftDungeons: LocalGameState.None_,
+        }
+        self.was_game_launched_task: asyncio.Task = None
+        self.update_task: asyncio.Task = None
+        self.check_sizes_task: asyncio.Task = None
+        self.owned = []
+        self.multimc: multimc.MultiMCClient = None
+
+    def _authenticate(self):
+        return Authentication("mojang_user", "Mojang User")
 
     async def authenticate(self, stored_credentials=None):
-        self.store_credentials({'dummy': 'dummy'})
-        return Authentication(user_id='Minecraft_ID', user_name='Minecraft Player')
+        log.debug(f"stored_credentials: {stored_credentials}")
+        if stored_credentials is not None and misc.IS(
+            ["owned", "multimcpath"], IN=stored_credentials
+        ):
+            self.owned = json.loads(stored_credentials["owned"])
+            if stored_credentials["multimcpath"] != "null":
+                self.multimc = multimc.MultiMCClient(stored_credentials["multimcpath"])
+            return self._authenticate()
+        return misc.get_next_step("Select Owned Games", 720, 720, "page1")

```

Comment:
One more thing I've found while testing old plugin - new version transition.

GOG user faces a unknown window (that has Galaxy-like CSS) on Galaxy start, that is unfair for Galaxy UX guys... Some users won't be interested in it at all and they will blame Galaxy for this "WTF" experience.

I have 2 solutions, you @urwrstkn8mare choose:
- (easy one) display ConnectionLost for all users. They need to click "Connect" again - now we have explicit. Technically: raise `InvalidCredentials` from `authenticate` if there is no `owned` in `stored_credentials`
- (nice one) `misc.get_next_step("Minecraft integration: new update", 720, 720, "migration_1.0")`. Page "zero" have only a few things:
    - Info that Minecraft integration has been updated and now need configuration.
    - (optionally) list of new features, but very short
    - buttons: Proceed (main button, will go to page1) | Logout (raise InvalidCredentials) | Skip (either to run the same page zero on next restart, or to function without new features (only Mincraft game owned??)

Reply:
> that is unfair for Galaxy UX guys

Sorry, I just used Galaxy-like CSS for consistency. I didn't mean to be unfair to anyone. 

> you @urwrstkn8mare choose

I think I would choose the nice one because first, it is nice :), and second it doesn't force users on to the new features/integration giving them a choice to stay with what they have. However, to implement it'll have to wait approximately a week or two. 

Sorry about that, I'll try to get it done as soon as possible.

Reply:
@urwrstkn8mare OK good choice :) If the ability to stay in old flow is problematic for you in terms of time or whatever - feel free to go with the "easy one". Having those new features in general has more value than one-time inconvenience for users. If you have more questions for me - ask!

Reply:
Thanks!

Reply:
Please see https://github.com/TouwaStar/Galaxy_Plugin_Minecraft/pull/13